### PR TITLE
feat: version config

### PR DESCRIPTION
**Pull request by andrii-codefresh**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

File: controller/state.go [hunk 1]
```diff
@@ -207,7 +208,12 @@ func (m *appStateManager) getRepoObjs(app *v1alpha1.Application, sources []v1alp
 			RefSources:         refSources,
 			ProjectName:        proj.Name,
 			ProjectSourceRepos: proj.Spec.SourceRepos,
-			VersionConfig:      apiclient.GetVersionConfig(),
+			ApplicationIdentity: &apiclient.ApplicationIdentity{
+				Runtime:   "runtime",
```
Comment (dmelnyk-octo) [inline controller/state.go]: why are these values hardcoded here?

File: pkg/version_config_manager/codefresh_client.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+			"namespace": app.Namespace,
+			"name":      app.Name,
+		},
+	}
+
+	return sendGraphQLRequest(client.config, query)
+}
+
+// sendGraphQLRequest function to send the GraphQL request and handle the response
+func sendGraphQLRequest(config Config, query GraphQLQuery) (*ApplicationConfiguration, error) {
```
Comment (dmelnyk-octo) [inline pkg/version_config_manager/codefresh_client.go]: is this a duplication of codefresh client from pkg/codefresh?

File: pkg/codefresh/client.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+func NewCodefreshClient(cfConfig *CodefreshConfig) *CodefreshClient {
+	return &CodefreshClient{
+		cfConfig: cfConfig,
+		httpClient: &http.Client{
+			Timeout: 30 * time.Second,
+		},
+	}
+}
+
+func (cc *CodefreshClient) Send(ctx context.Context, appName string, event *events.Event) error {
```
Comment (dmelnyk-octo) [inline pkg/codefresh/client.go]: Right now this client has 2 public methods.
One of them serves generic purpuse of sending any payload to the /2.0/api/events.
Second one is very specific and and submits specific graphql query to /2.0/api/graphql.

---
I propose to make this client have 2 generic methods (just rename existing) – `SendEvent` and `SendGraphQL`.
`GetApplicationConfiguration` and more other methods in the future can be implemented as part of other type `CodefreshQLInterface` or something like this:
```go
func NewCodefreshQL(cfClient *CodefreshConfig) *CodefreshQLInterface {
	return &CodefreshQLInterface{
		cfClient: cfClient
	}
}
```

Let me know WDYT.

File: server/application/application.go [hunk 4]
```diff
@@ -507,7 +507,12 @@ func (s *Server) GetManifests(ctx context.Context, q *application.ApplicationMan
 			EnabledSourceTypes: enableGenerateManifests,
 			ProjectName:        proj.Name,
 			ProjectSourceRepos: proj.Spec.SourceRepos,
-			VersionConfig:      apiclient.GetVersionConfig(),
+			ApplicationIdentity: &apiclient.ApplicationIdentity{
+				Runtime:   "runtime",
```
Comment (dmelnyk-octo) [inline server/application/application.go]: why are these values hardcoded here?

File: util/argo/argo.go [hunk 5]
```diff
@@ -773,7 +773,12 @@ func verifyGenerateManifests(
 			NoRevisionCache:    true,
 			HasMultipleSources: hasMultipleSources,
 			RefSources:         refSources,
-			VersionConfig:      apiclient.GetVersionConfig(),
+			ApplicationIdentity: &apiclient.ApplicationIdentity{
+				Runtime:   "runtime",
```
Comment (dmelnyk-octo) [inline util/argo/argo.go]: why are these values hardcoded here?

File: pkg/version_config_manager/codefresh_client.go (hunk 2 above)
Comment (andrii-codefresh) [reply pkg/version_config_manager/codefresh_client.go]: this client will be removed

File: pkg/codefresh/client.go [hunk 6]
```diff
@@ -19,27 +19,25 @@ type CodefreshConfig struct {
...
 
-type codefreshClient struct {
+type CodefreshClient struct {
 	cfConfig   *CodefreshConfig
 	httpClient *http.Client
 }
 
-type CodefreshClient interface {
+type CodefreshClientInterface interface {
 	Send(ctx context.Context, appName string, event *events.Event) error
```
Comment (dmelnyk-octo) [inline pkg/codefresh/client.go]: Rename to SendEvent

File: pkg/codefresh/client.go [hunk 7]
```diff
@@ -19,27 +19,25 @@ type CodefreshConfig struct {
...
+	SendGraphQLRequest(query GraphQLQuery) (*json.RawMessage, error)
```
Comment (dmelnyk-octo) [inline pkg/codefresh/client.go]: Rename to SendGraphQL

File: pkg/codefresh/api_graphql_requests.go [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+import (
+	"encoding/json"
+)
+
+type CodefreshGraphQLRequests struct {
+	client CodefreshClientInterface
+}
+
+type CodefreshGraphQLRequestsInterface interface {
```
Comment (dmelnyk-octo) [inline pkg/codefresh/api_graphql_requests.go]: CodefreshGraphQLRequestsInterface -> CodefreshGraphQLInterface

**Review by dmelnyk-octo — APPROVED** [review]

**Pull request merged**